src/main/controllers/RespondentApplicationDetailsController.ts

```diff
@@ -52,9 +52,66 @@ export default class RespondentApplicationDetailsController {
       }
     }
 
+    let claimantResponseDocDownload = undefined;
+
+    if (selectedApplication.value?.respondCollection && selectedApplication.value?.respondCollection.length) {
+      for (let i = selectedApplication.value?.respondCollection.length - 1; i >= 0; i--) {
+        if (
+          selectedApplication.value?.respondCollection[i].value.from === 'Claimant' &&
+          selectedApplication.value?.respondCollection[i].value.supportingMaterial !== undefined
+        ) {
+          claimantResponseDocDownload =
+            selectedApplication.value?.respondCollection[i].value.supportingMaterial[0].value.uploadedDocument;
+        }
+      }
+    }
+
+    if (claimantResponseDocDownload) {
+      try {
+        await getDocumentAdditionalInformation(claimantResponseDocDownload, req.session.user?.accessToken);
+      } catch (err) {
+        logger.error(err.message);
+        return res.redirect('/not-found');
+      }
+    }
+
+    const claimantResponseDocDownloadLink = createDownloadLink(claimantResponseDocDownload);
+
+    let decisionContent = undefined;
+    const decisionDocDownload = [];
+    const decisionDocDownloadLink = [];
+
+    if (selectedApplication.value?.adminDecision && selectedApplication.value?.adminDecision.length) {
+      for (let i = selectedApplication.value?.adminDecision.length - 1; i >= 0; i--) {
+        if (selectedApplication.value?.adminDecision[i].value.responseRequiredDoc !== undefined) {
+          decisionDocDownload[i] = selectedApplication.value?.adminDecision[i].value.responseRequiredDoc;
+        }
+      }
+    }
+
+    if (decisionDocDownload.length > 0) {
+      for (let i = decisionDocDownload.length - 1; i >= 0; i--) {
+        if (decisionDocDownload[i]) {
+          try {
+            await getDocumentAdditionalInformation(decisionDocDownload[i], req.session.user?.accessToken);
+          } catch (err) {
+            logger.error(err.message);
+            return res.redirect('/not-found');
+          }
+          decisionDocDownloadLink[i] = createDownloadLink(decisionDocDownload[i]);
+        }
+      }
+      decisionContent = getTseApplicationDecisionDetails(selectedApplication, translations, decisionDocDownloadLink);
+    }
```

Comment:
Could these segments be split out into their own method? This method is over 100 lines long, seems like its complexity would be high

Reply:
Moved claimant response doc download to helper